base/src/main/scala/clazz/Applicative.scala

```diff
@@ -2,8 +2,16 @@ package scato
 package clazz
 
 abstract class Applicative[F[_]] {
-  def apply: Apply[F]
+
   def pure[A](a: A): F[A]
+  def ap[A, B](fa: F[A])(f: F[A => B]): F[B]
+
+  def map[A, B](ma: F[A])(f: (A) => B): F[B] = ap(ma)(pure(f))
+
+  final val apply: Apply[F] = new Apply[F] {
```

Comment:
I think those kind of default implementation should not be final.

I can think of cases where we would like to give a reference instead, what do you think?